src/octoprint/util/comm.py

```diff
@@ -1299,6 +1318,7 @@ def _getGcodeScript(self, scriptName, replacements=None):
                 "printer_profile": self._printerProfileManager.get_current_or_default(),
                 "last_position": self.last_position,
                 "last_temperature": self.last_temperature.as_script_dict(),
+                "last_fanspeed": self.last_fanspeed,
```

Comment:
This would try to dump a record object into the placeholder, not the contained value. Same below. But since we should switch to an int anyhow (see above), this then will work.